var/spack/repos/builtin/packages/hipsparse/package.py

```diff
@@ -157,7 +138,11 @@ class Hipsparse(CMakePackage):
         "5.3.0",
         "5.3.3",
         "5.4.0",
+        "5.4.3",
     ]:
+        depends_on("rocm-cmake@%s:" % ver, type="build", when="@" + ver)
+        depends_on("hip@" + ver, when="@" + ver)
+        depends_on("rocsparse@" + ver, when="@" + ver)
         depends_on("rocprim@" + ver, when="@" + ver)
```

Comment:
This can be further simplified. rocprim is not a direct dependency of hipsparse. It's merely a build dependency of rocsparse, so it doesn't need to be mentioned in the hipsparse package recipe at all. I would just drop the `depends_on("rocprim",  ...)` and merge the two loops into one.

Reply:
Thanks ..I have made the changes